keras_cv/utils/preprocessing.py

```diff
@@ -94,13 +94,45 @@ def blend(image1: tf.Tensor, image2: tf.Tensor, factor: float) -> tf.Tensor:
     return tf.clip_by_value(temp, 0.0, 255.0)
 
 
-def parse_factor(param, min_value=0.0, max_value=1.0, param_name="factor", seed=None):
+def parse_factor(
+    param,
+    min_value=0.0,
+    max_value=1.0,
+    param_name="factor",
+    seed=None,
+    range_mode="non-negative",
+):
+    """A factor parameter parsing and validation function.
+
+    Arguments:
+        param: A single number, a tuple of two numbers or a `keras_cv.FactorSampler`.
+        min_value: Minimum allowed `param` value.
+        max_value: Maximum allowed `param` value.
+        param_name: Name of the passed `param`. Used to provide clear error message
+            when provided `param` has incorrect value.
+        seed: Integer. Used to create a random seed.
+        range_mode: Method used to interpret value of param if a single number is
+            provided. Supported values are `"non-negative"` which results in param
+            bounds `(min_value, param)` and `"symmetric"` with bounds `(-param, param)`.
+
+    Returns: `keras_cv.ConstantFactorSampler` or `keras_cv.UniformFactorSampler`
+
+    """
+    allowed_range_modes = {"non-negative", "symmetric"}
+    if range_mode not in allowed_range_modes:
+        raise ValueError(
+            f"`range_mode` should be one of {allowed_range_modes}. "
+            f"Got range_mode={range_mode}"
+        )
     if isinstance(param, core.FactorSampler):
         return param
 
-    if isinstance(param, float) or isinstance(param, int):
+    if isinstance(param, (float, int)) and range_mode == "non-negative":
```

Comment:
Calling this mode "non-negative" seems to assume that min_value is 0, which is not enforced by this API. I can't think of a better name for this, though, and I like the idea of including this as an option when parsing factors.

@LukeWood WDYT

Reply:
Yeah that sounds reasonable to me